src/tests/ftest/dfuse/daos_build.py

```diff
@@ -49,45 +52,91 @@ def test_daos_build(self):
-        dist = distro.linux_distribution()
-        if dist[0] == 'openSUSE Leap':
-            scons = 'scons'
-
         # Create a pool, container and start dfuse.
         self.add_pool(connect=False)
         self.add_container(self.pool)
 
         daos_cmd = self.get_daos_command()
-        daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
-                                    attr='dfuse-data-cache', val='off')
-
-        daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
-                                    attr='dfuse-attr-time', val='60s')
-
-        daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
-                                    attr='dfuse-dentry-time', val='60s')
 
-        daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
-                                    attr='dfuse-ndentry-time', val='60s')
+        cont_attrs = OrderedDict()
+
+        cache_mode = self.params.get('name', '/run/dfuse/*')
+
+        # How long to cache things for, if caching is enabled.
+        cache_time = '30m'
+        build_time = 240
+
+        if cache_mode == 'writeback':
+            cont_attrs['dfuse-data-cache'] = 'on'
+            cont_attrs['dfuse-attr-time'] = cache_time
+            cont_attrs['dfuse-dentry-time'] = cache_time
+            cont_attrs['dfuse-ndentry-time'] = cache_time
+            build_time = 15
+        elif cache_mode == 'writethrough':
+            cont_attrs['dfuse-data-cache'] = 'on'
+            cont_attrs['dfuse-attr-time'] = cache_time
+            cont_attrs['dfuse-dentry-time'] = cache_time
+            cont_attrs['dfuse-ndentry-time'] = cache_time
+        elif cache_mode == 'metadata':
+            cont_attrs['dfuse-data-cache'] = 'off'
+            cont_attrs['dfuse-attr-time'] = cache_time
+            cont_attrs['dfuse-dentry-time'] = cache_time
+            cont_attrs['dfuse-ndentry-time'] = cache_time
+        elif cache_mode == 'nocache':
+            cont_attrs['dfuse-data-cache'] = 'off'
+            cont_attrs['dfuse-attr-time'] = '0'
+            cont_attrs['dfuse-dentry-time'] = '0'
+            cont_attrs['dfuse-ndentry-time'] = '0'
+        else:
+            self.fail('Invalid cache_mode: {}'.format(cache_mode))
+
+        for key, value in cont_attrs.items():
+            daos_cmd.container_set_attr(pool=self.pool.uuid, cont=self.container.uuid,
+                                        attr=key, val=value)
 
         self.start_dfuse(self.hostlist_clients, self.pool, self.container)
 
         mount_dir = self.dfuse.mount_dir.value
         build_dir = os.path.join(mount_dir, 'daos')
 
-        cmds = ['git clone https://github.com/daos-stack/daos.git {}'.format(build_dir),
+        intercept = self.params.get('use_intercept', '/run/intercept/*', default=False)
+
+        remote_env = OrderedDict()
+        remote_env['PATH'] = '{}:$PATH'.format(os.path.join(mount_dir, 'venv', 'bin'))
```

Comment:
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string